### PR TITLE
Example: CINIC classification with Lance Dataset

### PR DESCRIPTION
**Pull request by viplismism**
@AyushExel @tanaymeh

### CODE REVIEW

File: community-examples/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## CNN Model with CINIC-10 Lance image dataset
+
+### Overview
+Today we are going to demonstrate how to train a Convolutional Neural Network (CNN) model using the CINIC-10 image dataset in the Lance format. The [CINIC-10](https://paperswithcode.com/dataset/cinic-10) dataset consists of 10 classes of images, including airplanes, automobiles, birds, cats, deer, dogs, frogs, horses, ships, and trucks.
```
Comment (AyushExel) [inline community-examples/README.md]: "This example demonstrates how to train ..."

**Comment by AyushExel** [comment]
Screenshot 2024-04-21 at 6 16 18
@vipul-maheshwari I think converting to df and operating on it is not necessary here. Instead just accept the path of the lance dataset, load it on initialisation and when called while training/validating just read the batch with offset?

**Review by AyushExel — APPROVED** [review]

**Comment by tanaymeh** [comment]
Why is the model accuracy at 15% @vipul-maheshwari?

And the loss is barely decreasing over the epochs which doesn't really show the point. Can you improve the model?

**Review by tanaymeh — CHANGES_REQUESTED** [review]
Please improve the training and testing accuracy and make sure the loss decreases.

**Comment by vipul-maheshwari** [comment]
The main goal for using lance image dataset to train a CNN model is to understand how using lance dataloader over vanilla dataloader can improve the time it takes to train one epoch, I have a report ready by the tomorrow to clarify more about this comparison.. 

tldr : The same script with lance dataloaders are taking almost half the time to train over an epoch as compared to the vanilla ones.. LANCE IS SPEEDY AS HELL 🚀

**Review by tanaymeh — APPROVED** [review]

**Comment by AyushExel** [comment]
Screenshot 2024-04-25 at 7 02 21 AM
@vipul-maheshwari `to_table()` turns lance dataset into pyarrow table, which is not lance format. 
The flow is to be something like this:
* DataLoader accepts lance dataset
* In the constructor you just load the lance dataset using lance.dataset(path)
* when getitem is called you simply access the data from lance format on that index
See this section from the docs - https://lancedb.github.io/lance/read_and_write.html#reading-lance-dataset
Screenshot 2024-04-25 at 7 04 52 AM
